### ISSUE TITLE
The training code provided in intel-extension-for-pytorch example still not working 

### ISSUE THREAD

**Issue by Jaja8877** [intel/intel-extension-for-pytorch#692]
Hi there,
I tried the Float32 code that intel provided in intel-extension-for-pytorch (https://intel.github.io/intel-extension-for-pytorch/cpu/2.3.0+cpu/tutorials/examples.html). but I encountered the issue ,which seems like the code wasn't working in the training process and instantly killed the training session itself.

the code is in below:
##############################################
import torch
import torchvision
import intel_extension_for_pytorch as ipex
from tqdm import tqdm
from torch.utils.data import DataLoader as DL

LR = 0.001
DOWNLOAD = False
DATA = 'datasets/cifar10/'

transform = torchvision.transforms.Compose([
    torchvision.transforms.Resize((224, 224)),
    torchvision.transforms.ToTensor(),
    torchvision.transforms.Normalize((0.5, 0.5, 0.5), (0.5, 0.5, 0.5))
])
train_dataset = torchvision.datasets.CIFAR10(
    root=DATA,
    train=True,
    transform=transform,
    download=DOWNLOAD,
)
train_loader = DL(
    dataset=train_dataset,
    batch_size=128
)

model = torchvision.models.resnet50()
criterion = torch.nn.CrossEntropyLoss()
optimizer = torch.optim.SGD(model.parameters(), lr=LR, momentum=0.9)
model.train()

model, optimizer = ipex.optimize(model, optimizer=optimizer)

for batch_idx, (data, target) in enumerate(tqdm(train_loader)):
    optimizer.zero_grad()
    output = model(data)
    loss = criterion(output, target)
    loss.backward()
    optimizer.step()
    print(batch_idx)

torch.save({
    'model_state_dict': model.state_dict(),
    'optimizer_state_dict': optimizer.state_dict(),
}, 'checkpoint.pth')

print("Execution finished")
##############################################

![intel-extension-for-pytorch issue](https://github.com/user-attachments/assets/2060e57e-fdaf-4255-b1c0-15665c1d182d)

My system info:
Dell Latitude 5550
Processor: Intel(R) Core(TM) Ultra 7 165U   1.70 GHz
RAM: 16.0 GB (15.5 GB available)
OS: Windows11 x64 P.S. But I used WSL (Ubuntu LTS 22.04) for enabling this package 

virtualenv:
Python version: 3.10.12
Torch version: 2.3.0+cpu
intel_extension_for_pytorch version: 2.3.100+cpu

A programming rookie first time to post an issue, if there is any required info. I should provided, please let me know, thanks.
